recipes-core/dbus/patches/fix-segfault-bus_connection_disconnected.patch

```diff
@@ -0,0 +1,13 @@
+--- a/bus/connection.c
++++ b/bus/connection.c
+@@ -302,7 +302,9 @@ bus_connection_disconnected (DBusConnect
+       if (mm != NULL)
+         bus_matchmaker_disconnected (mm, connection);
+ 
+-      _dbus_list_remove_link (&d->connections->monitors, d->link_in_monitors);
++      if (d->connections->monitors != NULL)
++        _dbus_list_remove_link (&d->connections->monitors, d->link_in_monitors);
```

Comment:
I think this will now leak link_in_monitors.  `_dbus_list_remove_link` does `_dbus_list_unlink` and `free_link`.  So maybe
```
if (d->connections->monitors != NULL)
    _dbus_list_unlink (&d->connections->monitors, d->link_in_monitors);

free_link (d->link_in_monitors);
d->link_in_monitors = NULL;
```

I looked at upstream dbus, and they haven't changed this code.  Why are we triggering it but not them?

Reply:
Yes, I checked upstream DBus too when debugging this and could not find a fix for that problem between 1.10.14 and 1.10.26. None of the patches applied in OpenXT seem relevant to that behavior either... This may be an assumption that is not honored, but I have not been able to figure out where.

- `_dbus_clear_list` will call `free_link` on each link of the list;
- `bus/connection.c:544: _dbus_list_clear (&connections->monitors)` will release `connections->monitors` list;
- Since `d->connections->monitors` in `bus_connection_disconnected` is `connections->monitors` in `bus_connections_unref`) and `d->link_in_monitors` is an element of that list it should be free'ed when run `bus_connection_disconnected` I think?

Reply:
Yes, your analysis looks correct.  So it seems like link_in_monitors should be NULLed when removing its entry from connections->monitors at bus/connection.c:544.  But that would mean bus_matchmaker_disconnected doesn't get run.